controllers/vspherecluster_controller.go

```diff
@@ -118,6 +137,7 @@ func AddClusterControllerToManager(ctx *context.ControllerManagerContext, mgr ma
 			&source.Channel{Source: ctx.GetGenericEventChannelFor(clusterControlledTypeGVK)},
 			&handler.EnqueueRequestForObject{},
 		).
+		WithEventFilter(predicates.ResourceIsNotExternallyManaged(reconciler.Logger)).
```

Comment:
shouldn't this be enough to ignore externally managed vsphereClusters? do we need the extra logic in the Cluster `Watches` above?

Reply:
Hi @gab-satchi  , this came up here https://kubernetes.slack.com/archives/C8TSNPY4T/p1620749340371300
so the issue is that we are watching a cluster then mapping it to a vsphereCluster,but the predicate is only applied on
the cluster, not the mapped vsphereCluster (and the cluster does not have the annotation).

Reply:
ah okay. thanks for the explanation!